### PR TITLE
fix: rename timeout

### PR DESCRIPTION
**Pull request by RohittCodes**
added a timeout and introduced a finally block

fixes: #295
/claim #295

### CODE REVIEW

**Review by seveibar — CHANGES_REQUESTED** [review]
this is fine, but can you switch this to use a react-query mutation instead? This type of logic should always use react-query

**Comment by RohittCodes** [comment]
this an api endpoint that failed. no changes made from my end.

**Comment by seveibar** [comment]
@RohittCodes use this

image


@Abse2001 can you confirm this fixes the underlying issue? I'm not super familiar w/ the problem

@RohittCodes nice refactor

**Comment by Abse2001** [comment]
@seveibar I will need few hours before I can test it Im going to collage.

**Comment by Abse2001** [comment]
It happens when you try to use a name that is not acceptable, eg (using spaces), after the fail message the dialog screen just stuck on renaming state, you can't change the name until you close it and open a new dialog.

**Comment by Abse2001** [comment]
@RohittCodes sorry for the delay I tested it and it looks fixed.
just fix the bun test and you are all good.

**Comment by RohittCodes** [comment]
@Abse2001 I actually don't know why the bun tests failed, I've not even tried changing anything in the api.

It's completely a frontend change.

**Comment by seveibar** [comment]
I think the bun tests are kind of flaky, just gotta rerun them (if that is the cause)

File: src/components/dialogs/rename-snippet-dialog.tsx [hunk 1]
```diff
@@ -24,31 +24,39 @@ export const RenameSnippetDialog = ({
...
         title: "Error",
         description: "Failed to rename the snippet. Please try again.",
         variant: "destructive",
       })
-    }
+    },
+  })
+
+  const handleRename = () => {
+    renameSnippetMutation.mutate()
```
Comment (seveibar) [inline src/components/dialogs/rename-snippet-dialog.tsx:59]: Just use an inline function here, these “handle” functions rarely increase readability especially when they are just proxying a mutation

**Review by seveibar — APPROVED** [review]

**Pull request merged**